File: app/router/http.go
package router

import (
	"github.com/aihop/gopanel/app/api"
	"github.com/aihop/gopanel/app/middleware"
	"github.com/aihop/gopanel/constant"
	"github.com/gofiber/fiber/v3"
)

func HttpRouter(r fiber.Router) {
	caddyRouter := r.Group("/http").
		Use(middleware.LocalOrJWT(constant.UserRoleAdmin))
	{
		caddyRouter.Get("/default/config", api.HttpDefaultRead)
		caddyRouter.Post("/default/update", api.HttpDefaultUpdate)
		caddyRouter.Post("/default/get", api.HttpDefaultGet)
		caddyRouter.Get("/default/list", api.HttpDefaultList)
		caddyRouter.Post("/default/check", api.HttpDefaultCheck)
		caddyRouter.Post("/default/delete", api.HttpDefaultDelete)
		caddyRouter.Post("/default/reload", api.HttpDefaultRestart)
		caddyRouter.Post("/default/stop", api.HttpDefaultStop)
		caddyRouter.Get("/default/status", api.HttpDefaultStatus)
		caddyRouter.Post("/default/resolve", api.HttpDefaultResolve)
	}
}

File: admin/src/api/modules/http.ts
import http from "@/api"

export const httpDefaultCheckAPI = (params: { domain: string }) => {
	return http.post<any>("/http/default/check", params)
}

export const httpDefaultResolveAPI = (params: { domain: string }) => {
	return http.post<any>("/http/default/resolve", params)
}

export const httpDefaultStatusAPI = () => {
	return http.get<any>("/http/default/status")
}

export const httpDefaultConfigAPI = () => {
	return http.get<any>("/http/default/config")
}

export const httpDefaultUpdateAPI = (params: { content: string }) => {
	return http.post<any>("/http/default/update", params)
}

export const httpDefaultListAPI = () => {
	return http.get<any>("/http/default/list")
}

export const httpDefaultDeleteAPI = (data: { domain: string }) => {
	return http.post<any>("/http/default/delete", data)
}

export const httpDefaultGetAPI = (data: { domain: string }) => {
	return http.post<any>("/http/default/get", data)
}

export const httpDefaultReloadAPI = () => {
	return http.post("/http/default/reload")
}

export const httpDefaultStopAPI = () => {
	return http.post("/http/default/stop")
}
